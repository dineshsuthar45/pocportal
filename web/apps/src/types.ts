import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
