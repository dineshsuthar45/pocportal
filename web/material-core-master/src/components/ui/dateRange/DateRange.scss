.date-range-root {
	width: 100%;

	* {
		font-family: Inter;
	}

	.tooltip-label {
		color: var(--m-colour-greyScale-500);
	}

	.selected {
		padding: 4px 4px 4px 0px;
		display: flex;
		cursor: pointer;
		position: relative;

		&:before {
			content: '';
			position: absolute;
			bottom: 0;
			width: 100%;
			border-bottom: 1px solid transparent;
		}

		&.disabled {
			pointer-events: none;
		}

		&:focus,
		&.menu-open {
			border-color: var(--m-colour-blue);
		}

		.selected-date {
			flex-grow: 1;
			text-overflow: ellipsis;
			white-space: nowrap;
			overflow: hidden;
		}
		.chevron {
			margin-left: auto;
		}
	}

	.calendar-menu {
		position: relative;

		.rdrDateRangePickerWrapper {
			position: absolute;
			box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
		}
	}
}
