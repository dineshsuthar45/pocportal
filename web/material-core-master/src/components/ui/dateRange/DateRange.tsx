import { useTheme } from "@mui/material";
import classNames from "classnames";
import { format } from "date-fns";
import * as Locales from "date-fns/locale";
import { useEffect, useRef, useState } from "react";
import { DateRangePicker } from "react-date-range";
import "react-date-range/dist/styles.css";
import "react-date-range/dist/theme/default.css";
import { useClickOutsideMultipleRefs } from "../../../hooks/";
import { MColours } from "../../../theme/colours";
import { MIconAngleDown, MIconAngleUp } from "../icons";
import { MFormControlLabel } from "../internal/formControlLabel";
import { MTooltip } from "../tooltip";
import "./DateRange.dark.scss";
import "./DateRange.light.scss";
import "./DateRange.scss";
import { DateRangeProps } from "./DateRange.types";

/**
 * Further properties can be found in react-date-range website at https://hypeserver.github.io/react-date-range/
 */
export const MDateRange = ({
  label,
  tooltipTitle = "",
  disabled = false,
  locale = "enGB",
  placeholder = "Select",
  range,
  onMenuChange,
  ...props
}: DateRangeProps) => {
  const theme = useTheme();
  const selectedRef = useRef<HTMLDivElement | null>(null);
  const dateFormat = "dd MMM yyyy";

  const [internalTooltipTitle, setInternalTooltipTitle] = useState(
    tooltipTitle || label || ""
  );
  const [showAlwaysTooltip, setShowAlwaysTooltip] = useState(false);
  const [initializationDone, setInitializationDone] = useState(false);
  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
  const [selectedDate, setSelectedDate] = useState("");
  const [color, setColor] = useState(
    theme.palette.mode === "light" ? MColours.blue : MColours.lightBlue
  );

  const calendarRef = useClickOutsideMultipleRefs<HTMLDivElement>(
    [selectedRef],
    () => setIsMenuOpen(false)
  );

  // todo it does work on second attempt
  // useEffect(() => {
  //   setRef(selectedRef.current);
  // }, [selectedRef.current]);

  // const [setRef, size] = useElementSize();

  // useEffect(() => {
  //   if (calendarRef.current) {
  //     (
  //       calendarRef.current.firstChild as HTMLDivElement
  //     ).style.width = `${size.width}px`;
  //   }
  // }, [calendarRef.current, size]);

  useEffect(() => {
    setInternalTooltipTitle(tooltipTitle || label || "");
  }, [tooltipTitle, label]);

  useEffect(() => {
    setShowAlwaysTooltip(!!tooltipTitle && tooltipTitle !== label);
  }, [label, tooltipTitle]);

  useEffect(() => {
    setInitializationDone(true);
  }, []);

  useEffect(() => {
    setColor(
      theme.palette.mode === "light" ? MColours.blue : MColours.lightBlue
    );
  }, [theme.palette.mode]);

  useEffect(() => {
    if (initializationDone) {
      const startDateFormatted = format(
        range.startDate ?? new Date(),
        dateFormat
      );
      const endDateFormatted = format(range.endDate ?? new Date(), dateFormat);

      setSelectedDate(`${startDateFormatted} - ${endDateFormatted}`);
    }
  }, [range.startDate, range.endDate, initializationDone]);

  useEffect(() => {
    onMenuChange && onMenuChange(isMenuOpen);
  }, [isMenuOpen, onMenuChange]);

  const handleSelectedClick = () => {
    setIsMenuOpen(!isMenuOpen);
  };

  return (
    <div className={classNames("date-range-root", theme.palette.mode)}>
      <MFormControlLabel
        label={label}
        tooltipProps={{
          title: internalTooltipTitle,
          showAlways: showAlwaysTooltip,
        }}
        disabled={disabled}
      >
        <>
          <div
            className={classNames("selected", {
              "menu-open": isMenuOpen,
              disabled,
            })}
            ref={selectedRef}
            onClick={handleSelectedClick}
          >
            <MTooltip title={selectedDate}>
              <span className="selected-date">
                {!selectedDate && (
                  <span className={classNames("placeholder", { disabled })}>
                    {placeholder}
                  </span>
                )}
                {selectedDate}
              </span>
            </MTooltip>
            <div className={classNames("chevron", { disabled })}>
              {!isMenuOpen && <MIconAngleDown />}
              {isMenuOpen && <MIconAngleUp />}
            </div>
          </div>

          <div ref={calendarRef} className="calendar-menu">
            {isMenuOpen && (
              <DateRangePicker
                color={color}
                rangeColors={[color]}
                ranges={[range]}
                locale={Locales[locale]}
                inputRanges={[]}
                months={2}
                direction="horizontal"
                {...props}
                onChange={props.onChange}
              />
            )}
          </div>
        </>
      </MFormControlLabel>
    </div>
  );
};

MDateRange.defaultProps = {
  tooltipTitle: "",
  disabled: false,
  locale: "enGB",
  placeholder: "Select",
};
