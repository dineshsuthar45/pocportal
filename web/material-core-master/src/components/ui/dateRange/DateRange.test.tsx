import { render } from "@testing-library/react";
import React from "react";
import { MDateRange } from "./DateRange";
import { DateRangeProps } from "./DateRange.types";

describe("DateRange", () => {
  const renderComponent = () => render(<MDateRange {...props} />);
  let props: DateRangeProps;

  beforeEach(() => {
    props = {
      label: "Date Picker",
      onChange: () => {},
      range: {
        startDate: new Date(2022, 11, 16),
        endDate: new Date(2022, 11, 16),
        key: "selection",
      },
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
