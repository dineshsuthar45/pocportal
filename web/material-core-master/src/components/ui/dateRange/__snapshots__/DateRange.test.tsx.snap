// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRange should render correctly 1`] = `
<div>
  <div
    class="date-range-root light"
  >
    <div
      class="m-form-control-label top"
    >
      <div
        aria-label=""
        class="m-tooltip-root tooltip-label tooltip-container"
        data-mui-internal-clone-element="true"
      >
        <label>
          Date Picker
        </label>
      </div>
      <div
        class="label-children"
      >
        <div
          class="selected"
        >
          <div
            aria-label=""
            class="m-tooltip-root tooltip-container"
            data-mui-internal-clone-element="true"
          >
            <span
              class="selected-date"
            >
              16 Dec 2022 - 16 Dec 2022
            </span>
          </div>
          <div
            class="chevron"
          >
            <svg
              fill="#6F7880"
              height="10"
              viewBox="0 0 10 10"
              width="10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m.5 3 4.1 4.1c.2.2.3.3.4.3s.2 0 .4-.2L9.6 3a.608.608 0 0 0 0-.8.608.608 0 0 0-.8 0L5 6.1 1.2 2.3a.483.483 0 0 0-.7 0 .483.483 0 0 0 0 .7Z"
              />
            </svg>
          </div>
        </div>
        <div
          class="calendar-menu"
        />
      </div>
    </div>
  </div>
</div>
`;
